crates/holochain/src/core/workflow/call_zome_workflow.rs

```diff
@@ -104,7 +104,7 @@ async fn call_zome_workflow_inner<'env, Ribosome: RibosomeT, C: CellConductorApi
     tracing::trace!(line = line!());
 
     let to_app_validate = {
-        let workspace = workspace_lock.read().await;
+        let mut workspace = workspace_lock.write().await;
```

Comment:
This is just a workspace lock, not an lmdb environment write transaction? or, if it is, it's ok, because despite saying it's a CallZomeWorkspace below, it's not actually making a zome call?

Reply:
It's just a write lock on the actual workspace not lmdb. I needed a write lock because the validation calls below need to update the cache from the calls they do.
The reason I need the call zome workspace in the first place is I need access to the data in the scratch data so that the validation can see the latest picture (basically imagine commit an entry then linking to it in the same zome call the validation needs to be able to see that entry to validate the link)